File: examples/ex13.m
main()
{
 array res0^1 <6>;
 array a^3 <0 3 4>;

 const array xnew^3 <2 3 4> = < 0 1 6 7 2 3 8 9 4 5 10 11 12
                                  13 18 19 14 15 20 21 16 17 22 23>;

 const array x^1 <1> = < 2 > ;
 const array j^1 <1> = <1> ;

 res0 = x drop ( x take ( j psi omega < 1 2 > xnew ) ) ;


}

File: examples/ex13.c
#include <stdlib.h>
#include "moalib.e"

main()

{
  double *offset0;
  int i0;
  int i1;
  int i2;
  double *shift;
  double _xnew[]={0.000000, 1.000000, 6.000000, 7.000000, 2.000000,
    3.000000, 8.000000, 9.000000, 4.000000, 5.000000,
    10.000000, 11.000000, 12.000000, 13.000000, 18.000000,
    19.000000, 14.000000, 15.000000, 20.000000, 21.000000,
    16.000000, 17.000000, 22.000000, 23.000000};
  double _x[]={2.000000};
  double _j[]={1.000000};
  int forall_i0[1];
  double _res0[6];

/*******
res0=x drop (x take (j psi  omega <1.000000 2.000000> xnew))
********/

  for (forall_i0[0]=2; forall_i0[0]<2+0; forall_i0[0]++) {
    shift=_res0+(forall_i0[0]+-2);
    offset0=_xnew+forall_i0[0]*12+1*4+0;
    for (i2=0; i2<4; i2++) {
      *(shift)= *(offset0);
      offset0+=1;
      shift+=0;
    }
  }


}


